packages/loopring_v3/contracts/aux/access/LoopringIOExchangeOwner.sol

```diff
@@ -128,6 +131,13 @@ contract LoopringIOExchangeOwner is SelectorBasedAccessManager, ERC1271, Drainab
         )
         private
     {
+        // Allocate memory to verify transactions that are approved here
+        bool[][] memory preApprovedTxs = new bool[][](blocks.length);
```

Comment:
With my improvement suggestion above, we don' have to use an im-memory preApprovedTxs any more, we can modify the data in `blocks` dirctly.

Reply:
The code gets a quite a bit more complicated if we do that (blocks without callbacks need all approved values false, gaps between callbacks need to be checked that the approved is false, all this needs to be checked so we have to careful with all possible scenarios), the temp arrays makes this easy and simple. 

But certainly possible, would have to measure it to see if it's worth the extra complexity. I still like not modifying the block data on the fly in the smart contracts if not necessary, but if it makes things better/more efficient than sure.

There reason I didn't just pass the data in a new separate array into submitBlocks next to the block data is that makes calling submitBlocks practically impossible to do efficiently. Now we just have to pass in a single bytes array which can be done very efficiently with fastCallAndVerify, even with large data. And so the data needs to be part of the initial bytes array somehow.

Reply:
Oh right, and the temp array also solves the problem of needing to lookup the tx in the aux data. There's no direct mapping between txIdx <-> auxIdx, so without the temp array we'd need some extra lookup data that needs to be sent onchain by the relayer or extra loops to find the necessary location (which could be quite expensive).